src/app/search/components/removeAllFiltersButton/removeAllFiltersButton.component.tsx

```diff
@@ -0,0 +1,27 @@
+"use client";
+
+import { useContext } from "react";
+
+import { FiltersContext } from "@/app/search/providers/filters/filters.provider";
+
+import styles from "./removeAllFiltersButton.module.css";
+
+export default function RemoveAllFiltersButtonComponent() {
+  const { filters, dispatchFilters } = useContext(FiltersContext);
+
+  return !!filters.query ||
+    filters.isVerified ||
+    !!filters.expertise ||
+    !!filters.gender ? (
+    <button
+      className={styles["remove"]}
+      onClick={() =>
+        dispatchFilters({
+          type: "removed_all",
+        })
+      }
+    >
+      حذف تمام فیلترها
+    </button>
+  ) : null;
```

Comment:
خیلی ناخوانا شده
بهتره اینجوری مواقع از if استفاده کنید

اگه از useMemo استفاده کنید هم که دیگه چه بهتر (مطابق ویدئو)

```suggestion
  const isEmpty =
    !!filters.query ||
    filters.isVerified ||
    !!filters.expertise ||
    !!filters.gender;

  if (!isEmpty) {
    return null;
  }

  return (
    <button
      className={styles["remove"]}
      onClick={() =>
        dispatchFilters({
          type: "removed_all",
        })
      }
    >
      حذف تمام فیلترها
    </button>
  );
```